src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,98 @@
+                             ProductMapper productMapper, SortOrderUtil sortOrderUtil) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+        this.sortOrderUtil = sortOrderUtil;
+    }
+
+    @PostMapping
+    @ApiOperation(value = "Create a new product")
+    public ProductResponseDto create(@RequestBody ProductRequestDto productRequestDto) {
+        Product product = productService.add(productMapper.toModel(productRequestDto));
+        return productMapper.toDto(product);
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toDto(productService.getById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ApiOperation(value = "Delete product")
+    public void deleteProduct(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation(value = "Update product")
+    public ProductResponseDto updateProduct(@PathVariable Long id,
+                                            @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productMapper.toModel(productRequestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.update(product));
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Get list of products with pagination")
+    public List<ProductResponseDto> findAll(@RequestParam(defaultValue = "20")
+                                                @ApiParam(value = "defaultValue = 20")
+                                                Integer count,
+                                            @RequestParam(defaultValue = "0")
+                                            @ApiParam(value = "defaultValue = 0") Integer page,
+                                            @RequestParam(defaultValue = "id")
+                                                @ApiParam(value = "defaultValue = id")
+                                                String sortBy) {
+
+        Sort sort = sortOrderUtil.sortBy(sortBy);
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAll(pageRequest).stream()
+                .map(productMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/by-price")
+    @ApiOperation(value = "Get list of products by price distance")
+    public List<ProductResponseDto> findAllByPriceBetween(@RequestParam
+                                                              @ApiParam(value = "Search from price")
+                                                              BigDecimal from,
+                                                          @RequestParam
+                                                          @ApiParam(value = "Search to price")
+                                                          BigDecimal to) {

```

Comment:
You forgot to add pagination for this endpoint:

![image](https://user-images.githubusercontent.com/67195687/210239299-08145ce5-e45a-482c-8f16-cead16361ff5.png)